packages/engine-server/src/workspace/utils.ts

```diff
@@ -102,6 +102,28 @@ export class WorkspaceUtils {
     return dendronWorkspaceFolders.filter(isNotUndefined);
   }
 
+  /**
+   * Check if a file is a dendron note (vs a regular file or something else entirely)
+   */
+  static async isDendronNote({
+    wsRoot,
+    vaults,
+    fpath,
+  }: { fpath: string } & WorkspaceOpts) {
+    // check if we have markdown file
+    if (!fpath.endsWith(".md")) {
+      return false;
+    }
+    // if markdown file, check if it is in a dendron vault
+    if (!WorkspaceUtils.isPathInWorkspace({ wsRoot, vaults, fpath })) {
+      return false;
+    }
+
+    // if markdown file, does it have frontmatter?
+    // this is a rough heuristic
+    return (await fs.readFile(fpath, { encoding: "utf8" })).startsWith("---");
```

Comment:
How about reading only the first part of the file here for further optimization in very large files? Looking it up a bit, looks like it is possible to read only some of the file using read streams. For example: https://stackoverflow.com/questions/70707646/reading-part-of-file-in-node

Reply:
updated. nice tip!